src/main/java/hudson/plugins/analysis/core/AnnotationDifferencer.java

```diff
@@ -40,7 +40,7 @@
     private static Set<FileAnnotation> difference(
             final Set<FileAnnotation> target, final Set<FileAnnotation> other) {
         Set<FileAnnotation> difference = Sets.newHashSet(target);
-        difference.removeAll(other);
+        difference.removeAll(Sets.newHashSet(other));
         return difference;
```

Comment:
I think it would be sufficient to run a test with a large amount of warnings and measure the system time. In getAnnotations an immutable set is returned.

I think we should not change code until we can make sure that it is really faster.